File: app/backend/src/main/java/es/nullpointers/eventvsmerida/entity/Categoria.java
package es.nullpointers.eventvsmerida.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * Entidad que representa una categoría de eventos.
 *
 * <p>Permite clasificar los eventos dentro de la aplicación,
 * por ejemplo música, teatro, deporte, cultura u otras categorías.</p>
 *
 * @author Eva Retamar
 * @author David Muñoz
 * @author Adrián Pérez
 */
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "\"Categoria\"")
public class Categoria {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @NotNull
    @Column(name = "nombre", nullable = false, length = Integer.MAX_VALUE)
    private String nombre;
} 

File: app/frontend/lib/models/categoria.dart
/// Modelo de datos que representa las categorías de los eventos en la app.
///
/// @author: Eva Retamar
/// @author: Adrián Pérez
/// @author: David Muñoz
class Categoria {
  final int id;
  final String nombre;

  Categoria({
    required this.id,
    required this.nombre,
  });

  Map<String, dynamic> toJson() => {
    'id': id,
    'nombre': nombre,
  };

  factory Categoria.fromJson(Map<String, dynamic> json) => Categoria(
    id: json['id'] ?? 12,
    nombre: json['nombre'] ?? '',
  );
}